### ISSUE TITLE
Merc fails the semantic analysis tests

### ISSUE THREAD

**Issue by pchapin** [cubesatlab/merc#19]
There are currently two classes in Merc that try to do semantic analysis: `SymbolTablePopulator` and `SemanticAnalyzer`. The `SemanticAnalyzer` class is essentially empty and its use is a no-op. However, the `SymbolTablePopulator` class fails to handle certain exceptions from the underlying `BasicSymbolTable` it is populating. This causes Merc to crash with an unhandled exception rather than report the error in the normal way.

The two classes should be merged, probably under the more precise name of `SemanticAnalyzer` and the matter of the unhandled exception should be fixed. After this is done, the semantic tests can be re-enabled (they are currently commented out so that SBT can complete the build process and not be stopped by the failing test).

**Comment by pchapin** [comment]
I added an exception handler in `SymbolTablePopulator` to catch the `DuplicateTypeNameException` from the symbol table and report on it. This fixes the failing semantic test so I re-enabled it. I also added some additional semantic tests for good measure.

For now, I kept the `SymbolTablePopulator` and `SemanticAnalyzer` classes separate. The `SemanticAnalyzer` class is now documented as a place holder for future expansion. Although it might never be necessary to have two classes, as the number of semantic checks increase, it might prove useful/desirable to split some of them off into a separate pass over the parse tree. Let's keep that infrastructure in place for now. I did remove the distracting commented-out code from `SemanticAnalyzer`. It appeared to be very old since it didn't make much sense anyway (or has been superseded by `SymbolTablePopulator`).

Note the comment in `SymbolTablePopulator` toward the bottom of the `visitBasic_TypeDef` method. It could be that we should be reporting errors from the symbol table in some other way than by throwing an exception.
